### PR TITLE
routes and controllers added

### PR DESCRIPTION
**Pull request by AhsanAkbar252**
Added routes and controller files

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  root 'welcome#home'
+  resources :users
```
Comment (suleman-uzair) [inline config/routes.rb:3]: if want to have a custom action e.g `search`, how would i define it ?

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+  resources :products
```
Comment (suleman-uzair) [inline config/routes.rb:4]: what actions does `resources` keyword automatically defines ?

File: app/controllers/products_controller.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
+    @product.save
+    redirect_to products_path
+  end
+
```
Comment (suleman-uzair) [inline app/controllers/products_controller.rb:14]: add path comment above action like this

# GET products/:id

for show action

File: config/routes.rb (hunk 1 above)
Comment (AhsanAkbar252) [reply config/routes.rb:3]: get 'users/search', to: 'users#search'

File: config/routes.rb (hunk 2 above)
Comment (AhsanAkbar252) [reply config/routes.rb:4]: new,create,update,index,show,edit,destroy

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  # GET products/:id
```
Comment (suleman-uzair) [inline app/controllers/products_controller.rb:15]: add path for all controller actions

File: config/routes.rb (hunk 1 above)
Comment (suleman-uzair) [reply config/routes.rb:4]: we can define it through resources as well, use that

File: app/views/products/new.html.erb [hunk 5]
```diff
@@ -1,8 +1,12 @@
...
-  <%= f.label :name %>
-  <%= f.text_field :name %><br>
-
-  <%= f.label :price %>
-  <%= f.text_field :price %><br>
-  <%= f.submit 'Create Product' %>
-<% end %>
\ No newline at end of file
+<%= form_with model: @product do |form|%>
+<%= form.label :query, 'Search for:' %>
```
Comment (suleman-uzair) [inline app/views/products/new.html.erb:2]: indentation issues

**Pull request merged**